engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -221,10 +221,41 @@ namespace dmGameSystem
  * ```
  */
 
+struct SetTextureAsyncRequest
+{
+    dmhash_t                   m_PathHash;
+    lua_State*                 m_LuaState;
+    dmScript::LuaCallbackInfo* m_CallbackInfo;
+    TextureResource*           m_TextureResource;
+    dmBuffer::HBuffer          m_Buffer;
+    int32_t                    m_BufferRef;
+    dmGraphics::HTexture       m_Texture;
+    HOpaqueHandle              m_Handle;
+    bool                       m_UseUploadBuffer;
+};
+
+struct CreateTextureResourceParams
+{
+    const char*                               m_Path;
+    dmhash_t                                  m_PathHash;
+    dmGameObject::HCollection                 m_Collection;
+    dmGraphics::TextureType                   m_Type;
+    dmGraphics::TextureFormat                 m_Format;
+    dmGraphics::TextureImage::Type            m_TextureType;
+    dmGraphics::TextureImage::TextureFormat   m_TextureFormat;
+    dmGraphics::TextureImage::CompressionType m_CompressionType;
+    dmBuffer::HBuffer                         m_Buffer;
+    uint32_t                                  m_Width;
+    uint32_t                                  m_Height;
+    uint32_t                                  m_MaxMipMaps;
+    uint32_t                                  m_TextureBpp;
+};
```

Comment:
Q: Do we want to create a script_resource_texture.cpp and move the texture functions and structs into it? this file is getting kinda big..

Reply:
I think i'll do this separately